shopping_mall/backend/ai/agent/executor.py

```
@@ -0,0 +1,652 @@
+        context: RequestContext | None = None,
+    ) -> AgentResult:
+        """에이전트 루프 실행. Primary 실패 시 Fallback으로 전환."""
+        ctx = context or RequestContext.build(user_id)
+        system_with_ctx = system + ctx.to_system_suffix()
+
+        try:
+            return await self._run_loop(
+                self.primary, db, user_message, user_id, session_id, history, system_with_ctx
+            )
+        except AgentUnavailableError as e:
+            logger.warning(f"Primary LLM 실패: {e}. Fallback 시도.")
```

Ruff (0.15.9)

[warning] Logging statement uses f-string

(G004)